public/app/features/transformers/utils.ts

```diff
@@ -64,7 +66,7 @@ export const numberOrVariableValidator = (value: string | number) => {
   if (!Number.isNaN(Number(value))) {
     return true;
   }
-  if (/^\$[A-Za-z0-9_]+$/.test(value)) {
+  if (value.match(variableRegex) !== null) {
```

Comment:
https://dev.to/dvddpl/why-is-my-regex-working-intermittently-4f4g

If we use the imported variable, the global flag will cause it to flip between true and false with `.test`. Match will work consistently.

Reply:
heh, yeah ran into this silent footgun recently: https://github.com/grafana/grafana/pull/98416/files

you can still use `variableRegex.test()`, but right afterwards rewind `variableRegex.lastIndex = 0`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex#rewinding_lastindex

Reply:
fixed!